backtrace-library/src/main/java/backtraceio/library/metrics/UniqueEvent.java

```diff
@@ -0,0 +1,62 @@
+package backtraceio.library.metrics;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import backtraceio.library.common.BacktraceStringHelper;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class UniqueEvent extends Event {
+
+    private final static transient String LOG_TAG = UniqueEvent.class.getSimpleName();
+
+    /**
+     * Unique events API spec requires unique events to be a JSON array, but we still treat it as a single string
+     */
+    @SerializedName("unique")
+    private final List<String> name;
+
+    UniqueEvent(String name) {
+        this(name, new HashMap<String, Object>());
+    }
+
+    UniqueEvent(String name, Map<String, Object> attributes) {
+        this(name, System.currentTimeMillis() / 1000, attributes);
+    }
+
+    UniqueEvent(String name, long timestamp, Map<String, Object> attributes) {
+        super(timestamp);
+        this.attributes = attributes;
+        this.name = new ArrayList<String>() {{
+            add(name);
+        }};
+    }
+
+    /**
+     * Unique events API spec requires unique events to be a JSON array, but we still treat it as a single string
+     */
+    @Override
+    public String getName() {
+        if (this.name != null && this.name.size() > 0) {
```

Comment:
probably you can use isNullOrEmpty method from BacktraceStringHelper

Reply:
Good point, will do!